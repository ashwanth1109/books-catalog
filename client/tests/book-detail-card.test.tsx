import { render } from "@testing-library/react";
import BookDetailCard from "../components/book-detail-card.component";

describe("book-detail-card.component.tsx", () => {
  it("should assert that dom snapshot is correct", () => {
    const { container } = render(
      <BookDetailCard
        loading={true}
        title="Title"
        description="Description"
        year="2003"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
