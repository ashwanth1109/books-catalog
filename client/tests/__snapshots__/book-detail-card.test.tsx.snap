// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`book-detail-card.component.tsx should assert that dom snapshot is correct 1`] = `
<div>
  <div
    class="ant-card ant-card-loading ant-card-bordered"
    style="width: 500px;"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-loading-content"
      >
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-22"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-8"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-15"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-18"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-13"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-9"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
        <div
          class="ant-row"
          style="margin-left: -4px; margin-right: -4px;"
        >
          <div
            class="ant-col ant-col-4"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-3"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
          <div
            class="ant-col ant-col-16"
            style="padding-left: 4px; padding-right: 4px;"
          >
            <div
              class="ant-card-loading-block"
            />
          </div>
        </div>
      </div>
    </div>
    <ul
      class="ant-card-actions"
    >
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <span
            aria-label="shopping-cart"
            class="anticon anticon-shopping-cart"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="shopping-cart"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 00-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 100 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 00-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 00-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 00-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 01-31.6 31.6z"
              />
            </svg>
          </span>
        </span>
      </li>
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <span
            aria-label="book"
            class="anticon anticon-book"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="book"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M832 64H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V96c0-17.7-14.3-32-32-32zm-260 72h96v209.9L621.5 312 572 347.4V136zm220 752H232V136h280v296.9c0 3.3 1 6.6 3 9.3a15.9 15.9 0 0022.3 3.7l83.8-59.9 81.4 59.4c2.7 2 6 3.1 9.4 3.1 8.8 0 16-7.2 16-16V136h64v752z"
              />
            </svg>
          </span>
        </span>
      </li>
      <li
        style="width: 33.333333333333336%;"
      >
        <span>
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
        </span>
      </li>
    </ul>
  </div>
</div>
`;
